proto/tendermint/abci/types.proto

```diff
@@ -127,17 +127,31 @@ message RequestApplySnapshotChunk {
 }
 
 message RequestPrepareProposal {
-  // block_data is an array of transactions that will be included in a block,
+  // the modified transactions cannot exceed this size.
+  int64 max_tx_bytes = 1;
+  // txs is an array of transactions that will be included in a block,
   // sent to the app for possible modifications.
-  // applications can not exceed the size of the data passed to it.
-  tendermint.types.Data block_data = 1;
-  // If an application decides to populate block_data with extra information, they can not exceed this value.
-  int64 block_data_size = 2;
+  repeated bytes            txs                  = 2;
```

Comment:
[question] can the protobuf field `2` be reused? Is it a breaking change if it is reused? Asking b/c prior to this PR field `2` was a in64 for block_data_size